File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        temp = []
        for i in range(len(nums)):
            temp.append([nums[i], i])
        temp.sort(reverse = True)
        s = set()
        for i in range(k):
            s.add(temp[i][1])
        ans = []
        for i in range(len(nums)):
            if i in s:
                ans.append(nums[i])
        return ans

File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        int n = nums.size();
        vector<int> arr = nums;
        for (int i = 0; i < n; i++)
        {
            pq.push(nums[i]);
        }
        vector<int> ans(k);
        for (int i = 0; i < n - k; i++)
        {
            arr.erase(find(arr.begin(), arr.end(), pq.top()));
            pq.pop();
        }
        for (int i = 0; i < k; i++)
        {
            ans[i] = arr[i];
        }
        return ans;
    }
};